packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -460,6 +468,18 @@ class NavigatorState extends State<Navigator> {
     return _history.length > 1 || _history[0].willHandlePopInternally;
   }
 
+  void didStartUserGesture() { config.observer?.didStartUserGesture(); }
```

Comment:
Changed this to a helper interface vended by the Route and Navigator, as we discussed.